### ISSUE TITLE
GenericWhitespace false positive for parameterized record pattern

### ISSUE THREAD

**Issue by igormukhin** [checkstyle/checkstyle#20505]
I have read the check documentation:
https://checkstyle.org/checks/whitespace/genericwhitespace.html#GenericWhitespace

I have tested this with Checkstyle 13.7.0.

## Problem

`GenericWhitespaceCheck` reports a violation for a valid parameterized record pattern where the generic type argument is immediately followed by the record pattern component list.

## Minimal reproducer

```java
record Pair<T>(T left, T right) {
}

class Test {
    void test(Object obj) {
        if (obj instanceof Pair<?>(var left, var right)) {
            left.toString();
        }
    }
}
```

## Config

```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="GenericWhitespace"/>
    </module>
</module>
```

## Actual behavior

Checkstyle reports:

```text
GenericWhitespace '>' should followed by whitespace. [GenericWhitespace]
```

The reported token is the `>` in:

```java
Pair<?>(var left, var right)
      ^
```

## Expected behavior

No violation.

The syntax is valid Java. `javac --release 25` accepts it. The `(` after `>` is the record pattern component list, not a place where whitespace should be required.

## Notes

This appears related to, but not fully covered by:

- https://github.com/checkstyle/checkstyle/issues/15008
- https://github.com/checkstyle/checkstyle/issues/14502
- https://github.com/checkstyle/checkstyle/pull/14529

Those issues/PRs appear to cover whitespace around generic record headers and spaced record-pattern examples like `Pair < ? > (...)`, but Checkstyle 13.7.0 still reports a false positive for the compact valid form `Pair<?>(...)`.

From inspecting `GenericWhitespaceCheck`, it has special handling for generics before methods, constructors, and record headers, but this parameterized `RECORD_PATTERN_DEF` shape still falls through to the generic-end follower check where `(` is treated as invalid.

**Comment by krishna5379chaudhary-eng** [comment]
Hi, I'd like to work on this issue. I believe the false positive is caused by `GenericWhitespaceCheck` not allowing `(` after `>` for parameterized record patterns. Could you please assign it to me?

**Comment by romani** [comment]
@krishna5379chaudhary-eng , we have no assignments, please start with good xxxxxx issues, to be onboarded smoothly.

@Sahitya0805 , please help community here, you are onboarded enough to handle it.

**Comment by sahitya0xsingh** [comment]
Thanks! I'll start working on a fix.
